manifests/params.pp

```diff
@@ -93,27 +93,27 @@
       # We use system packages except on EL7
       if versioncmp($facts['os']['release']['major'], '8') >= 0 {
         $passenger_ruby_package = undef
-        $plugin_prefix = 'rubygem-foreman_'
+        $_plugin_prefix = 'rubygem-foreman_'
```

Comment:
I think for both EL7 & EL8 we can nowadays use `foreman-plugin-` as a prefix. Puppet should support `Provides` and it would simplify matters. For example, the puppetdb plugin also follows the same convention (tasks appears to be `foreman-plugin-foreman-tasks`). Thoughts?

Reply:
Would surely make a great follow up PR? ;)

Reply:
But yes, puppet 5.5 already has the `virtual_packages` feature in `yum` and `dnf` providers, so it's probably safe to switch to that.

Reply:
An noting it down explicitly: we don't have a similar provides for Hammer plugins. We should, but that means we can't immediately depend on it.

Reply:
If we'd like to do that in the future, can an issue be filed to track it?

Reply:
Yes, I was about to open follow up issues for the things we talked about.

Reply:
https://projects.theforeman.org/issues/33095 - Use "Provides" for hammer plugins too
https://projects.theforeman.org/issues/33096 - Use virtual package names for Foreman plugins
https://projects.theforeman.org/issues/33097 - Use virtual package names for Hammer plugins

Reply:
Thanks! Should we file issues for the Foreman Proxy equivalents too while we're at it?

Reply:
Yeah, sounds good. I just didn't think of it as it's not part of this PR

Reply:
Shall I create them or will you? Just so we don't end up with duplicates.

Reply:
You. I'm away from the pc

Reply:
https://projects.theforeman.org/issues/33099 / https://projects.theforeman.org/issues/33100 / https://projects.theforeman.org/issues/33101